### PR TITLE
The language detection is now done in blocks

### PR DESCRIPTION
**Pull request by annando**
The default language detection has got problems when it comes to texts that are held in both latin and non latin languages. For that reason for a long time we already had a coding there that guesstimatet the main language of the post.

This is now done differently. Now the text is split into separate blocks. These blocks are then chedked individually. The result is then combined.

To avoid problems, the score is reduced by a factor that reflects the amount of characters in that language in the whole text.

Please also have got a look at the addon PR: https://git.friendi.ca/friendica/friendica-addons/pulls/1428

### CODE REVIEW

**Comment by MrPetovan** [comment]
PHPUnit log:
```
1) Friendica\Test\src\Module\Api\Twitter\Statuses\UpdateTest::testApiStatusesUpdate
Error: Class 'IntlChar' not found

/woodpecker/src/github.com/friendica/friendica/src/Model/Item.php:2075
/woodpecker/src/github.com/friendica/friendica/src/Model/Item.php:2036
/woodpecker/src/github.com/friendica/friendica/src/Model/Item.php:1995
/woodpecker/src/github.com/friendica/friendica/src/Model/Item.php:1264
/woodpecker/src/github.com/friendica/friendica/src/Module/Api/Twitter/Statuses/Update.php:181
/woodpecker/src/github.com/friendica/friendica/src/BaseModule.php:229
/woodpecker/src/github.com/friendica/friendica/src/Module/BaseApi.php:105
/woodpecker/src/github.com/friendica/friendica/tests/src/Module/Api/Twitter/Statuses/UpdateTest.php:59

2) Friendica\Test\src\Module\Api\Twitter\Statuses\UpdateTest::testApiStatusesUpdateWithHtml
Error: Class 'IntlChar' not found

/woodpecker/src/github.com/friendica/friendica/src/Model/Item.php:2075
/woodpecker/src/github.com/friendica/friendica/src/Model/Item.php:2036
/woodpecker/src/github.com/friendica/friendica/src/Model/Item.php:1995
/woodpecker/src/github.com/friendica/friendica/src/Model/Item.php:1264
/woodpecker/src/github.com/friendica/friendica/src/Module/Api/Twitter/Statuses/Update.php:181
/woodpecker/src/github.com/friendica/friendica/src/BaseModule.php:229
/woodpecker/src/github.com/friendica/friendica/src/Module/BaseApi.php:105
/woodpecker/src/github.com/friendica/friendica/tests/src/Module/Api/Twitter/Statuses/UpdateTest.php:81
```

File: src/Model/Item.php [hunk 1]
```diff
@@ -2064,6 +2063,10 @@ public static function getLanguageArray(string $body, int $count, int $uri_id =
 	 */
 	private static function splitByBlocks(string $body): array
 	{
+		if (class_exists('IntlChar')) {
```
Comment (MrPetovan) [inline src/Model/Item.php]: Doesn't this imply the `IntlChar` module is optional?

Comment (annando) [reply src/Model/Item.php]: Well, it isn't optional in the future. But without this check surely several installation would do some WSOD after install.

File: src/Core/Installer.php [hunk 2]
```diff
@@ -399,15 +397,25 @@ public function checkFunctions()
...
-				$returnVal = false;
-			}
+		} elseif (!function_exists('mysqli_connect') && class_exists('pdo') && !in_array('mysql', \PDO::getAvailableDrivers())) {
+			$status = false;
+			$help = DI::l10n()->t('Error: The MySQL driver for PDO is not installed.');
+			$returnVal = false;
 		}
 		$this->addCheck(DI::l10n()->t('PDO or MySQLi PHP module'), $status, true, $help);
 
+		// Acticate when IntlChar is installed in the check pipeline.
```
Comment (MrPetovan) [inline src/Core/Installer.php]: Wouldn't this benefit admins though, despite the check pipeline failing?
```suggestion
		// Uncomment when IntlChar is installed in the check pipeline.
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

**Comment by annando** [comment]
Question is how to enable the module in the pipeline system.

File: src/Core/Installer.php (hunk 2 above)
Comment (annando) [reply src/Core/Installer.php]: Of course I would like to activate it - but then the pipeline would fail all the time.

**Comment by MrPetovan** [comment]
Isn't the Continuous Integration pipeline building from the Friendica Docker image?

**Comment by MrPetovan** [comment]
Nevermind, we have our own Friendica CI images: https://hub.docker.com/u/friendicaci which are build based on the content of this repository: https://git.friendi.ca/friendica/friendica-docker-ci

That's where the module should be added.

**Comment by annando** [comment]
PR added: https://git.friendi.ca/friendica/friendica-docker-ci/pulls/13

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**